tests/components/nws/test_weather.py

```diff
@@ -152,46 +115,68 @@
     assert forecast is None
 
 
-async def test_error_station(hass):
+async def test_error_station(hass, mock_simple_nws):
     """Test error in setting station."""
-    with patch("homeassistant.components.nws.weather.SimpleNWS") as mock_nws:
-        instance = mock_nws.return_value
-        instance.station = "ABC"
-        instance.set_station.side_effect = aiohttp.ClientError
-        instance.update_observation.return_value = mock_coro()
-        instance.update_forecast.return_value = mock_coro()
-        instance.observation = None
-        instance.forecast = None
-        await async_setup_component(hass, "weather", MINIMAL_CONFIG)
 
-        state = hass.states.get("weather.abc")
-        assert state is None
+    instance = mock_simple_nws.return_value
+    instance.set_station.side_effect = aiohttp.ClientError
+
+    assert await async_setup_component(hass, nws.DOMAIN, MINIMAL_CONFIG) is True
+    await hass.async_block_till_done()
+
+    assert hass.states.get("weather.abc_hourly") is None
+    assert hass.states.get("weather.abc_daynight") is None
 
 
-async def test_error_observation(hass, caplog):
+async def test_error_observation(hass, mock_simple_nws, caplog):
     """Test error during update observation."""
-    with patch("homeassistant.components.nws.weather.SimpleNWS") as mock_nws:
-        instance = mock_nws.return_value
-        instance.station = "ABC"
-        instance.set_station.return_value = mock_coro()
-        instance.update_observation.side_effect = aiohttp.ClientError
-        instance.update_forecast.return_value = mock_coro()
-        instance.observation = None
-        instance.forecast = None
-        await async_setup_component(hass, "weather", MINIMAL_CONFIG)
+    instance = mock_simple_nws.return_value
+    instance.update_observation.side_effect = aiohttp.ClientError
 
-    assert "Error updating observation from station ABC" in caplog.text
+    assert await async_setup_component(hass, nws.DOMAIN, MINIMAL_CONFIG)
+    await hass.async_block_till_done()
 
+    instance.update_observation.side_effect = None
 
-async def test_error_forecast(hass, caplog):
+    future_time = dt_util.utcnow() + timedelta(minutes=15)
+    async_fire_time_changed(hass, future_time)
+    await hass.async_block_till_done()
+
+    assert "Error updating observation" in caplog.text
+    assert "Success updating observation" in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5329046)